### PR TITLE
Migrate wasm to proto

### PR DESCRIPTION
**Pull request by sahith-narahari**
- Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/gaia/blob/master/CONTRIBUTING.md#pr-targeting))

- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Wrote tests
- [ ] Updated relevant documentation (`docs/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: x/wasm/keeper/keeper_test.go [hunk 1]
```diff
@@ -183,21 +184,21 @@ func TestInstantiate(t *testing.T) {
...
-	gasBefore := ctx.GasMeter().GasConsumed()
+	//gasBefore := ctx.GasMeter().GasConsumed()
 
 	// create with no balance is also legal
 	addr, err := keeper.Instantiate(ctx, contractID, creator, initMsgBz, "demo contract 1", nil)
 	require.NoError(t, err)
 	require.Equal(t, "cosmos18vd8fpwxzck93qlwghaj6arh4p7c5n89uzcee5", addr.String())
 
-	gasAfter := ctx.GasMeter().GasConsumed()
-	require.Equal(t, uint64(24823), gasAfter-gasBefore)
```
Comment (sahith-narahari) [inline x/wasm/keeper/keeper_test.go:194]: I couldn't figure out how this expected value is calculated, but it's changing after migration and causing this test to fail

Comment (ethanfrey) [reply x/wasm/keeper/keeper_test.go:194]: yeah, this happens a lot with changes... modifying storage access by one byte, or many other aspects will change this. The point is the same code run on every dev machine and the CI should be the same. I just update this and if it works locally and in CI, then :+1:

**Comment by ethanfrey** [comment]
Looks good, I will merge this into the other PR, then do the final checks.

File: x/wasm/handler.go [hunk 2]
```diff
@@ -60,22 +60,23 @@ func handleStoreCode(ctx sdk.Context, k Keeper, msg *MsgStoreCode) (*sdk.Result,
 		return nil, err
 	}
 
-	codeID, err := k.Create(ctx, msg.Sender, msg.WASMByteCode, msg.Source, msg.Builder)
+	codeID, err := k.Create(ctx, msg.Sender, msg.WasmByteCode, msg.Source, msg.Builder)
 	if err != nil {
 		return nil, err
 	}
 
-	events := filterMessageEvents(ctx.EventManager())
```
Comment (ethanfrey) [inline x/wasm/handler.go:68]: It seems you are missing the `filterMessageEvents`, which may be causing this other error.

File: x/wasm/keeper/keeper_test.go (hunk 1 above)
Comment (ethanfrey) [reply x/wasm/keeper/keeper_test.go:194]: I will update and fix them

File: x/wasm/handler.go (hunk 2 above)
Comment (sahith-narahari) [reply x/wasm/handler.go:68]: Yeah, this maybe the reason. Will modify this and see

Comment (ethanfrey) [reply x/wasm/handler.go:68]: fixed in my last commits

**Pull request merged**